nixos/modules/tasks/filesystems/exfat.nix

```
@@ -4,8 +4,10 @@ with lib;
 
 {
   config = mkIf (any (fs: fs == "exfat") config.boot.supportedFilesystems) {
-
-    system.fsPackages = [ pkgs.exfat ];
-
+    system.fsPackages = with config.boot.kernelPackages; (if kernelOlder "5.7" then [
```

I'd just use
```suggestion
    system.fsPackages = if config.boot.kernelPackages.kernelOlder "5.7" then [
```

The `with ;` expression has some problems, see https://nix.dev/anti-patterns/language#with-attrset-expression, and in this case it's not really necessary.